spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkTruncateFunction.java

```diff
@@ -0,0 +1,377 @@
+import org.junit.Test;
+
+public class TestSparkTruncateFunction extends SparkTestBaseWithCatalog {
+  public TestSparkTruncateFunction() {}
+
+  @Before
+  public void useCatalog() {
+    sql("USE %s", catalogName);
+  }
+
+  @Test
+  public void testTruncateTinyInt() {
+    Assert.assertEquals((byte) 0, scalarSql("SELECT system.truncate(10, 0Y)"));
+    Assert.assertEquals((byte) 0, scalarSql("SELECT system.truncate(10, 1Y)"));
+    Assert.assertEquals((byte) 0, scalarSql("SELECT system.truncate(10, 5Y)"));
+    Assert.assertEquals((byte) 0, scalarSql("SELECT system.truncate(10, 9Y)"));
+    Assert.assertEquals((byte) 10, scalarSql("SELECT system.truncate(10, 10Y)"));
+    Assert.assertEquals((byte) 10, scalarSql("SELECT system.truncate(10, 11Y)"));
+    Assert.assertEquals((byte) -10, scalarSql("SELECT system.truncate(10, -1Y)"));
+    Assert.assertEquals((byte) -10, scalarSql("SELECT system.truncate(10, -5Y)"));
+    Assert.assertEquals((byte) -10, scalarSql("SELECT system.truncate(10, -10Y)"));
+    Assert.assertEquals((byte) -20, scalarSql("SELECT system.truncate(10, -11Y)"));
+
+    // Check that different widths can be used
+    Assert.assertEquals((byte) -2, scalarSql("SELECT system.truncate(2, -1Y)"));
+
+    // Check that tinyint types are allowed for the width
+    Assert.assertEquals((byte) 0, scalarSql("SELECT system.truncate(5Y, 1Y)"));
+
+    Assert.assertEquals(
+        "Null input should return null",
+        null,
+        scalarSql("SELECT system.truncate(2, CAST(null AS tinyint))"));
+  }
+
+  @Test
+  public void testTruncateSmallInt() {
+    Assert.assertEquals((short) 0, scalarSql("SELECT system.truncate(10, 0S)"));
+    Assert.assertEquals((short) 0, scalarSql("SELECT system.truncate(10, 1S)"));
+    Assert.assertEquals((short) 0, scalarSql("SELECT system.truncate(10, 5S)"));
+    Assert.assertEquals((short) 0, scalarSql("SELECT system.truncate(10, 9S)"));
+    Assert.assertEquals((short) 10, scalarSql("SELECT system.truncate(10, 10S)"));
+    Assert.assertEquals((short) 10, scalarSql("SELECT system.truncate(10, 11S)"));
+    Assert.assertEquals((short) -10, scalarSql("SELECT system.truncate(10, -1S)"));
+    Assert.assertEquals((short) -10, scalarSql("SELECT system.truncate(10, -5S)"));
+    Assert.assertEquals((short) -10, scalarSql("SELECT system.truncate(10, -10S)"));
+    Assert.assertEquals((short) -20, scalarSql("SELECT system.truncate(10, -11S)"));
+
+    // Check that different widths can be used
+    Assert.assertEquals((short) -2, scalarSql("SELECT system.truncate(2, -1S)"));
+
+    // Check that short types are allowed for the width
+    Assert.assertEquals((short) 0, scalarSql("SELECT system.truncate(5S, 1S)"));
+
+    Assert.assertEquals(
+        "Null input should return null",
+        null,
+        scalarSql("SELECT system.truncate(2, CAST(null AS smallint))"));
+  }
+
+  @Test
+  public void testTruncateInt() {
+    Assert.assertEquals(0, scalarSql("SELECT system.truncate(10, %d)", 0));
```

Comment:
Why are the ints put into SQL using a format string? None of the other tests do that.

Reply:
Hold over from a previous version. Updated!